app/javascript/utilities/AxiosClient.js

```diff
@@ -0,0 +1,16 @@
+import axios from 'axios';
+import { getSessionToken } from '@shopify/app-bridge-utils';
+
+const instance = axios.create();
+
+const interceptor = instance.interceptors.request.use(
+  function (config) {
+    return getSessionToken(window.app)
```

Comment:
`getSessionToken` is async, isn't it? Do you need to `await` here?

Do axios interceptors support `async` middleware?

Reply:
I looked it up, they do! This is awesome, thanks Nabeel!

Reply:
Yep! The function passed into an interceptor expects that a `config` object is ultimately returned at some point to pass into subsequent requests.

In this example we're just promising that it will happen eventually.